### DISCUSSION TITLE
There was an error generating the updated file: Cannot read property 'monthlyPriceInCents' of null ❌

### DISCUSSION THREAD

**Discussion by jaydrogers** [JamesIves/github-sponsors-readme-action#498] [Q&A] [answered]
Thanks for putting this project together, this is really cool!

I am running into an issue with my project and I am hoping to see if I am doing everything correctly.

# ⚠️ Error I am experiencing
```
Error: There was an error generating the updated file: Cannot read property 'monthlyPriceInCents' of null ❌
```
#### Screenshot of Actions Log (with debugging output)
<img width="1312" alt="Screen Shot 2022-10-03 at 11 37 29" src="https://user-images.githubusercontent.com/3174134/193631627-77259664-dbbd-458b-8e24-8897e1091142.png">

# 👨‍🔬 Configuration Settings

### Sponsorship Type: `organization`
I am attempting to access the sponsors for Server Side Up (https://github.com/sponsors/serversideup) through the user "serversideup-bot" (https://github.com/serversideup-bot)

### My README file
https://github.com/serversideup/spin/blob/main/README.md

### My WorkFlow
https://github.com/serversideup/spin/blob/main/.github/workflows/sponsors.yml

### Personal Access Token Scopes
![Screenshot 2022-10-03 at 11-31-02 New Personal Access Token](https://user-images.githubusercontent.com/3174134/193633539-2f0fcb05-a4ef-4af3-8f60-b21023c3718b.png)

# 🤔 Current Results
Currently, I am not seeing anything appear on my README: https://github.com/serversideup/spin

<img width="959" alt="Screen Shot 2022-10-03 at 11 45 34" src="https://user-images.githubusercontent.com/3174134/193633252-13707856-8056-47b9-8dd4-201408eb8d0f.png">

# 🙏 Expected Results
- I would like anyone who is defined as `supporters` to appear on our README automatically

# ❓Questions
I saw the discussion on #450, but I have a few questions:
1. I assume using a dedicated "bot user" is the best method? I am unsure of other methods for creating access tokens as an organization
2. Are my configurations correct? Or is this possibly a bug in the action itself?

**Comment by JamesIves** [comment, marked as answer]
This looks correct to me. Does the bot user have org admin access? Seems like this is likely a permission error, although it's odd that it's throwing on the price in cents.

I'm also curious if you're able to run a graphql query in the GitHub api explorer from this user for sponsorship data and see if it properly returns - https://docs.github.com/en/graphql/overview/explorer

**Comment by jaydrogers** [comment]
Changing the bot user to an "Owner" on the team, helped me get one step closer!

### What it did
It was able to get me sponsors data! ([see the diff →](https://github.com/serversideup/spin/commit/0dcb67b75b8a3007dcaddb3dea83b4476013f2c0)).

When I did this, I accidentally had my settings set to `minimum`:
https://github.com/serversideup/spin/blob/8ae2d7460a9bb7c495db85645624a3a68a98625d/.github/workflows/sponsors.yml#L16-L17

### Where I am at now
- If I change it to `maximum`, it is returning my "fallback"


<img width="984" alt="Screen Shot 2022-10-04 at 13 07 10" src="https://user-images.githubusercontent.com/3174134/193893867-51a17d18-c676-4483-a91e-861d6e078f72.png">

### View my current code
Everything is live here: https://github.com/serversideup/spin

### Next steps
I am very grateful for your help on this awesome project! Let me know if I should open a new issue or not? I didn't know if the `maximum` issue was related to the first error that I was getting.

**Comment by JamesIves** [comment]
The `maximum` and `minimum` values let you specify a range. If you want to show _all_ sponsors, don't specify either at all. However if you want users to pay a certain amount before showing their avatar, then you'll want to set a `minimum` value (note: it's priced in cents not dollars). The idea of the `maximum` value is to cap sponsors at a certain amount so you can separate them by pricing tiers, there's a [guide on this in the readme here](https://github.com/JamesIves/github-sponsors-readme-action#separating-by-sponsorship-tier-).

If none of your sponsors match the criteria specified then the fallback will be shown. In your case your sponsor has paid more than 5 cents, and as a result the fallback is being shown.  Keep in mind you can run the sponsorship job multiple times in a given workflow, so you can have fallbacks for different pricing tiers should you choose too.

**Reply by jaydrogers** [reply]
Thanks @JamesIves! Sorry for missing that in the docs. Changing my maximum to `500` solved my problem 😅

This project is incredible. Thank you so much for your work and your friendly & timely support 🙌
